### DISCUSSION TITLE
Help needed with rssiFactor & measuredPower

### DISCUSSION THREAD

**Discussion by acidShrooms2** [mKeRix/room-assistant#653] [Q&A] [answered]
Hi there,

I have some questions regarding rssiFactor & meassuredPower and would be happy if someone coult help. 

1. I have a setup of multiple Raspi 2 (with usb BT Adapter), Raspi 3 and Raspi Zero W. The problem is, that the Zero Ws BT Adapter seems to be way stronger than the others. Am I right, that rssiFactor is used to compensate for this? If yes, how do I determine the right value and how is it taken into account? Is it just meassured rssi * rssiFactor?
2.  The differnt signal strenghts seem to apply for the devices I'm tracking. While some show a distance of 0 when placed next to the beacon, others meassure up to -24. As I understood meassuredPower is used to balance this out. But I have no idea where I could get that Value from and how I need to set it to correct the beacons meassurment.

**Comment by mKeRix** [comment, marked as answer]
I assume we are talking about Bluetooth Low Energy? If so, your general understanding is correct.

1. Exactly, for antennas on the Pis you'd use the `rssiFactor` option to balance out the measurements. It takes the measured RSSI and multiplies it with the factor, e.g. `-54 * 0.9 = -48,6` for an `rssiFactor` of 0.9.  I suppose a good way to calculate this is to look at the measured RSSI (not distance!) in the API when one of your device is exactly 1m away and then match the instances so that this value is always roughly the same for each instance.
2. `measuredPower` is the value used to estimate the distance from measured RSSI. It should be the RSSI value that you see when the device is exactly 1m away from the measuring instance. You could combine this with the tuning for question 1, then you only have to go through the measuring once. So if you see a measured RSSI of `-54` for the device when it is exactly 1m away from the instance you'd set the `measuredPower` to be `-54` for that device.

**Reply by acidShrooms2** [reply]
Thanks for the explanation, going to try that right away. Would you recommend to just choose one Raspi as reference and adjust the others to match it?

**Comment by acidShrooms2** [comment]
So I tried this out now and it seems that `measuredPower` works as expected. If I place my devices 1m away from any Raspi it shows a distance of 1 (+|- 0.2). But when I increase the distance, the Values are off again. I think it might have something to do with the different transmitting powers of the raspis. Thats what I wanted to correct using the `rssiFactor`. But setting the value doesn't seem to make any difference at all. To test it I've set `rssiFactor` to 0.4 on one of the raspis but the meassured rssi as the meassured distance stay the same after restarting the ra service. I think for the meassured rssi value thats the expected behaviour but shouldn't the distance change after setting the `rssiFactor`? I've attached my BLE settings just to be sure nothing is wrong with it.

```
...
bluetoothLowEnergy:
  hciDeviceId: 1
  rssiFactor: 0.4
  allowlist:
    - 'xxxxx1'
    - 'xxxxx2'
  tagOverrides:
    xxxxx1:
      name: Watch
      measuredPower: -57.4324
    xxxxx2:
      name: Phone BLE
      measuredPower: -83.0071
...
```

**Reply by mKeRix** [reply]
RSSI and estimated distance should definitely change when you configure an `rssiFactor`. Are you running the latest version of room-assistant (v2.16.0)? The factor feature is only available since that version. You can check the currently installed version with `room-assistant --help`.

Note that distance estimation will grow to be more inaccurate the further devices are away or if there are some objects in-between. This is by nature of how BLE distance estimation works.

**Reply by acidShrooms2** [reply]
Oh yes that was it. I was testing it on my home-assistant instance and didn't install the update. Thank you so much for your effort.
